src/transformStateWithClones.js

```diff
@@ -6,8 +6,37 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrayStateConditions = [];
+  let stateNew = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties': {
+        stateNew = { ...stateNew, ...action.extraData };
+        break;
+      }
+
+      case 'removeProperties': {
+        stateNew = { ...stateNew };
+
+        for (const key of action.keysToRemove) {
+          delete stateNew[key];
+        }
+        break;
+      }
+
+      case 'clear': {
+        stateNew = {};
+        break;
+      }
+    }
+
+    arrayStateConditions.push(stateNew);
```

Comment:
To avoid mutating the `stateNew` object that is being pushed into `arrayStateConditions`, you should push a copy of `stateNew` instead. This ensures that subsequent actions do not alter the previous states in the history.